### ISSUE TITLE
[CEP-47] User Permission shows wrong user display name (in special cases)

### ISSUE THREAD

**Issue by evsa0317** [codeclou/customfield-editor-plugin#47]
Hi Team,
We are using Customfield Editor Plugin (1.3 on Production and 1.7 on test server)
When we open the below url to add/remove users
https://<baseurl>/secure/CustomFieldEditorPlugin-AdminEditCustomField.jspa?customfieldid=<ID>
Actual result:
![image](https://user-images.githubusercontent.com/36727151/42157622-d0c90cfe-7e0b-11e8-9034-63b8316e6674.png)
Expected result:
![image](https://user-images.githubusercontent.com/36727151/42157790-537aecc6-7e0c-11e8-8e3d-e84a877e1fb3.png)

Please let me know why there is a discrepancy in username and Fullname.

Thanks.

**Comment by clouless** [comment]
Hi,

This is due to simple search mechanism that searches by username and internal JIRA user Search also searches in the lastname of users.

In your special case you seem to have a user with username `usera` for `User X`
And you seem to have another user with username `???` for `User A`.

Since the actual names displayed are just info you do not have to worry that the wrong permissions are given. **The permissions are given by the username** (not the real name).

Possible solutions for your problem would be to have a username policy like `lastname.firstname` to avoid such conflicts.

-----

I will see if I can make the search stricter to just search by username.
I will further investigate and see if I can reproduce the bug. If it is reproducible I will fix it in the next release.

**Comment by clouless** [comment]
Currently this is not reproducible for me.

![image](https://user-images.githubusercontent.com/12599965/42176719-3b7e8dc6-7e2a-11e8-9c22-041170d5f1f5.png)

@evsa0317 can you Use Google Chrome and send me the data of the `searchUser` XHR query from the Developer Tools network tab, when you search for `usera`. If this is sensitive data, you can send it to DISCLOSED.

thx

**Comment by evsa0317** [comment]
Hi,
PFB XHR query resolut when I search with usera:

```
[
   {
      "text":"irso0417 (I. Soo)",
      "id":"irso0417"
   },
   {
      "text":"sorosoo (E. Soo)",
      "id":"sorosoo"
   },
]
```

FYI: PFB Usernames and Fullnames:
irso0417  - I. Soo
sorosoo    - E. Soo

_(updated issue and anonymized names - GDPR)_

**Comment by clouless** [comment]
Thx for the response. I will try to recreate your scenario and try to see where the error is.

**Comment by clouless** [comment]
hi @evsa0317 I can not confirm that this is a bug. I will release a bugifx release with version 1.7.1 at the end of the week.

**Comment by clouless** [comment]
# ✅  Implemented BugFix in 1.7.1

<img width="1160" alt="01-user-exists" src="https://user-images.githubusercontent.com/12599965/42464049-32bd3fc4-83a8-11e8-8b61-c8da23391ca5.png">
<img width="1090" alt="02-user-deleted" src="https://user-images.githubusercontent.com/12599965/42464050-32d91082-83a8-11e8-93b5-4e63f3539c0b.png">

**Comment by clouless** [comment]
## ✅ Added to Administrator Guide:

https://codeclou.io/customfield-editor-plugin/1.7.1/administrator-guide/

![image](https://user-images.githubusercontent.com/12599965/42525907-ab20dc92-8474-11e8-8f56-1425bc7c8886.png)

**Comment by clouless** [comment]
## ✅ Just released plugin version 1.7.1 on Marketplace

**Comment by clouless** [comment]
@evsa0317 please report back to me if the error still occurs with plugin version 1.7.1 😃
